nevergrad/optimization/optimizerlib.py

```diff
@@ -2615,7 +2653,20 @@ def _select_optimizer_cls(self) -> base.OptCls:
 
 
 @registry.register
-class NGOpt(NGOpt14):
+class NGOpt15(NGOpt14):  # Also known as NGOpt12H_nohyperopt
+    def _select_optimizer_cls(self) -> base.OptCls:
+        if (
+            self.budget is not None
+            and self.budget < self.dimension**2 * 2
```

Comment:
```suggestion
            and self.budget < self.dimension**2 * 2
            and not self.has_noise
```

Reply:
The version of optimizerlib.py that you commented on is outdated. In the latest one, the condition on has_noise is already there.